### PR TITLE
Preventing from using the same canvas for multiple render()

### LINKED ISSUE #5636: Error when the same canvas is used for two render operations
To illuminate the common mistake: throw an exception if the canvas/context is used to start the rendering when other one with the same context is in progress

... and probably modify hello work example to e.g. print console message to indicate when rendering is finished.

### PR DESCRIPTION
**Pull request by yurydelendik**
Attempt to make life less confusing (#8476)

Fixes #5636

### CODE REVIEW

File: src/display/api.js [hunk 1]
```diff
@@ -2125,13 +2126,25 @@ var InternalRenderTask = (function InternalRenderTaskClosure() {
...
+    this._canvas = params.canvasContext.canvas;
   }
 
   InternalRenderTask.prototype = {
 
     initializeGraphics:
         function InternalRenderTask_initializeGraphics(transparency) {
 
+      if (this._canvas) {
+        console.log(this._canvas);
```
Comment (timvandermeij) [inline src/display/api.js]: Nit: remove the left-over debug print

File: test/unit/api_spec.js [hunk 2]
```diff
@@ -1041,6 +1041,32 @@ describe('api', function() {
...
+        viewport,
+      });
+      var renderTask2 = page.render({
+        canvasContext: canvasAndCtx.context,
+        viewport,
+      });
+
+      Promise.all([
+        renderTask1.promise,
+        renderTask2.then(() => {
```
Comment (Snuffleupagus) [inline test/unit/api_spec.js]: Nit: `renderTask2.promise.then` for consistency?

File: src/display/api.js [hunk 3]
```diff
@@ -2125,13 +2126,24 @@ var InternalRenderTask = (function InternalRenderTaskClosure() {
...
 
   InternalRenderTask.prototype = {
 
     initializeGraphics:
         function InternalRenderTask_initializeGraphics(transparency) {
 
+      if (this._canvas) {
+        if (canvasInRendering.has(this._canvas)) {
+          throw new Error(
+            'Cannot use the same canvas during multiple render() operations.' +
```
Comment (Snuffleupagus) [inline src/display/api.js]: Nit: add space at the end of this string, to avoid it reading `... operations.Use different...`.

**Review by Snuffleupagus — APPROVED** [review]
Looks good, thank you for the patch!

**Pull request merged**